### PR TITLE
fuchsia.ui.pointer.TouchSource implementation for fuchsia platform view

### PR DESCRIPTION
**Pull request by jaeheon**
Implement fuchsia.ui.pointer for fuchsia platform view. Makes touch input independent of Scenic's legacy API.

*List which issues are fixed by this PR. You must list at least one issue.*
https://github.com/flutter/flutter/issues/88620

## Pre-launch Checklist

- [x ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x ] I listed at least one issue that this PR fixes in the description above.
- [x ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x ] I updated/added relevant documentation (doc comments with `///`).
- [x ] I signed the [CLA].
- [x ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by jaeheon** [comment]
David and Darren - just wanted to get this out there. It's ready for a round of review, but note some rough edges below. I'll work on these remaining items next week. 

Some todos, as listed in the commit log:
- implement a toggle for gfx
- validate trace events gathered in a real trace
- manual testing against ToT product
- platform view unittest for logical->physical translation

**Review by arbreng — APPROVED** [review]
LGTM but as discussed offline there are 2 pieces left:
1.  Add the logic to toggle this input path via a flag
2. Add missing platform_view_unittest cases

File: shell/platform/fuchsia/flutter/pointer_delegate.cc [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+using fup_EventPhase = fuchsia::ui::pointer::EventPhase;
+using fup_TouchEvent = fuchsia::ui::pointer::TouchEvent;
+using fup_TouchIxnStatus = fuchsia::ui::pointer::TouchInteractionStatus;
+using fup_TouchResponse = fuchsia::ui::pointer::TouchResponse;
+using fup_TouchResponseType = fuchsia::ui::pointer::TouchResponseType;
+using fup_ViewParameters = fuchsia::ui::pointer::ViewParameters;
+
+namespace {
+void MaybeIssueTraceEvent(const fup_TouchEvent& event) {
+  if (event.has_trace_flow_id()) {
```
Comment (mikaelpessa) [inline shell/platform/fuchsia/flutter/pointer_delegate.cc:34]: Isn't having a trace flow id an API guarantee? It should be...

File: shell/platform/fuchsia/flutter/pointer_delegate.cc [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+        FML_DCHECK(view_parameters_.has_value()) << "API guarantee";
+        auto events = CreatePointerDraft(event, view_parameters_.value());
+        if (buffer_.count(ixn) > 0) {
+          InsertIntoBuffer(std::move(events), &buffer_[ixn]);
+        } else {
+          InsertIntoBuffer(std::move(events), &to_client);
+        }
+        // For this simple client, always claim we want the gesture.
+        response.set_response_type(fup_TouchResponseType::YES);
```
[the remark this answers is not in the document]
Comment (jaeheon) [reply shell/platform/fuchsia/flutter/pointer_delegate.cc]: Done

**Comment by jaeheon** [comment]
Status update - ready for review. Some changes in planning:

- This patch adds the TouchSource code, but does *not* hook it up to either GFX (legacy scenic) or Flatland. 
- I have verified that this patch does not affect the existing (legacy) touch dispatch, via touch-input-test on fuchsia.git.
- Next step is to write client code for MouseSource, and hook it up to Flatland (ie my next patch, not this patch).
- *Later* I will try hook up TouchSource/MouseSource to GFX via a toggle flag, but that is not critical path work, so punting to future date (tracked by fxbug.dev/85125, and marked in engine.cc).

File: shell/platform/fuchsia/flutter/tests/fakes/touch_source.h [hunk 3]
```diff
@@ -0,0 +1,60 @@
+// Copyright 2021 The Fuchsia Authors. All rights reserved.
```
Comment (jaeheon) [inline shell/platform/fuchsia/flutter/tests/fakes/touch_source.h:1]: fix copyright block

**Comment by jaeheon** [comment]
David, PTAL regarding weak-ptr fixup. If looks good, i'll merge it in.

**Review by arbreng — APPROVED** [review]

**Pull request merged**